src/main/java/fr/quatrevieux/araknemu/game/fight/Fight.java

```diff
@@ -336,4 +338,12 @@ public void destroy() {
         teams.clear();
         map.destroy();
     }
+
+    public void addInvocation(InvocationFighter invocation, FightCell cell) {
```

Comment:
To use lambda for adding the invocation on the fight :

Create the interface for the factory. It must have only one method
```java
@FunctionalInterface
public interface InvocationFighterFactory {
    public Fighter create(int id);
}
```

Use as parameter of the method, and call create() with the id (the code is the same, but with the lambda call, and return the created fighter) :
```java
    public Fighter addInvocation(InvocationFighterFactory factory, FightCell cell) {
        final Fighter invocation = factory.create(fighters().stream().mapToInt(Fighter::id).min().getAsInt() - 1);

        invocation.joinFight(this, cell);
        turnList.currentFighter().team().join(invocation);
        turnList.add(invocation);

        invocation.init();

        return invocation;
    }
```